internal/serve/httphandler/assets_handler.go

```
@@ -154,7 +171,15 @@ func (c AssetsHandler) DeleteAsset(w http.ResponseWriter, r *http.Request) {
 	httpjson.Render(w, asset, httpjson.JSON)
 }
 
-func (c AssetsHandler) handleUpdateAssetTrustlineForDistributionAccount(ctx context.Context, assetToAddTrustline *txnbuild.CreditAsset, assetToRemoveTrustline *txnbuild.CreditAsset) error {
+func (c AssetsHandler) handleUpdateAssetTrustlineForDistributionAccount(
+	ctx context.Context, assetToAddTrustline *txnbuild.CreditAsset, assetToRemoveTrustline *txnbuild.CreditAsset, distributionAccount schema.TransactionAccount,
+) error {
+	// Non-native Stellar distribution accounts will not require asset trustlines to be managed on our end. This is
+	// technically unreachable from the endpoint entry points, but we will still check for this case here.
+	if !distributionAccount.IsStellar() {
+		return nil
```

I think we should return an error here in case this is not a Stellar account.